js/packages/common/src/components/ConnectButton/index.tsx

```diff
@@ -1,42 +1,51 @@
-import { Button, Dropdown, Menu } from "antd";
-import { ButtonProps } from "antd/lib/button";
-import React from "react";
-import { useWallet } from './../../contexts/wallet';
+import { Button, Dropdown, Menu } from 'antd';
+import { ButtonProps } from 'antd/lib/button';
+import React, { useCallback } from 'react';
+import { useWallet } from '@solana/wallet-adapter-react';
+import { useWalletModal } from '../../contexts';
 
-export interface ConnectButtonProps extends ButtonProps, React.RefAttributes<HTMLElement> {
+export interface ConnectButtonProps
+  extends ButtonProps,
+    React.RefAttributes<HTMLElement> {
   allowWalletChange?: boolean;
 }
 
-export const ConnectButton = (
-  props: ConnectButtonProps
-) => {
-  const { connected, connect, select, provider } = useWallet();
+export const ConnectButton = (props: ConnectButtonProps) => {
   const { onClick, children, disabled, allowWalletChange, ...rest } = props;
 
-  // only show if wallet selected or user connected
+  const { wallet, connect, connected } = useWallet();
+  const { setVisible } = useWalletModal();
+  const open = useCallback(() => setVisible(true), [setVisible]);
 
-  const menu = (
-    <Menu>
-      <Menu.Item key="3" onClick={select}>Change Wallet</Menu.Item>
-    </Menu>
+  const handleClick = useCallback(
+    () => (wallet ? connect().catch(() => {}) : open()),
```

Comment:
This silently catches because there's already a global error handler on the context provider that notifies if there's a connection issue. Wallets that couldn't be found to detect to open a window with the website of the provider.